server/src/main/java/org/opensearch/cluster/routing/allocation/decider/NodeVersionAllocationDecider.java

```diff
@@ -58,15 +57,13 @@ public class NodeVersionAllocationDecider extends AllocationDecider {
 
     public static final String NAME = "node_version";
 
-    private final ReplicationType replicationType;
-
-    public NodeVersionAllocationDecider(Settings settings) {
-        replicationType = IndexMetadata.INDEX_REPLICATION_TYPE_SETTING.get(settings);
-    }
+    public NodeVersionAllocationDecider() {}
 
     @Override
     public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
         if (shardRouting.primary()) {
+            IndexMetadata indexMd = allocation.metadata().getIndexSafe(shardRouting.index());
+            final ReplicationType replicationType = IndexMetadata.INDEX_REPLICATION_TYPE_SETTING.get(indexMd.getSettings());
```

Comment:
I don't think it is the choice of right and wrong but whether it makes sense to have indices with mixed replication types on a node.
cc: @gbbafna @sachinpkale @ashking94 What do you think?